go/storage/mkvs/checkpoint/chunk.go

```diff
@@ -16,65 +18,213 @@ import (
 	"github.com/oasisprotocol/oasis-core/go/storage/mkvs/syncer"
 )
 
-func createChunk(
-	ctx context.Context,
-	tree mkvs.Tree,
-	root node.Root,
-	offset node.Key,
-	chunkSize uint64,
-	w io.Writer,
-) (
+type writerFactory interface {
+	next() (int, io.WriteCloser, error)
+}
+
+type seqChunker struct {
```

Comment:
Add documentation comments briefly describing what `seqChunker` and `parallelChunker` do.